simulation/model.py

```diff
@@ -0,0 +1,102 @@
+import datetime
+from collections import Counter
+from typing import Dict, Iterator
+
+import pandas as pd
+
+from abm import AgentBasedModel, UnorderedSpace
+from simulation.agents import Household
+from simulation.collectors import get_agent_collectors, model_collectors
+from simulation.constants import (
+    BuiltForm,
+    Epc,
+    HeatingSystem,
+    OccupantType,
+    PropertyType,
+)
+
+
+class CnzAgentBasedModel(AgentBasedModel):
+    def __init__(
+        self,
+        start_datetime,
+        step_interval,
+    ):
+        self.start_datetime = start_datetime
+        self.step_interval = step_interval
+        self.current_datetime = start_datetime
+
+        super().__init__(UnorderedSpace())
+
+    @property
+    def current_agent_count(self) -> Dict:
+        return Counter([agent.__class__ for agent in self.space.agents])
+
+    @property
+    def current_heating_system_count(self) -> Dict[HeatingSystem, int]:
+        return Counter(
+            [
+                agent.heating_system
+                for agent in self.space.agents
+                if isinstance(agent, Household)
+            ]
+        )
```

Comment:
I don't recognise these two properties from before. There's no tests either. Is it new?

I think these things would be better suited for model collectors.

We would know the number of agents anyway because agents aren't removed in the simulation.

Reply:
Not new, as per previous repo. Agreed, we can remove it.